### ISSUE TITLE
Case insensitive uniqueness validation is broken with Rails 5 & Postgres

### ISSUE THREAD

**Issue by nickrivadeneira** [rails/rails#27005]
Tested this with Sqlite and it worked fine, but with Postgres it was broken.
### Steps to reproduce

You'll need a database called `testapp_test` for these scripts:
[Failing executable test case for AR 5](https://gist.github.com/nickrivadeneira/952e052a718dabb9986ca902316274f5/b4a1ac74ac3750ae6ad1eedce1d2e2ba8c529722)
[Passing executable test case for AR 4.2](https://gist.github.com/nickrivadeneira/952e052a718dabb9986ca902316274f5/0ee46c99d327af54020c3f63c4d17148a10908a1)

### Expected behavior
AR creates a SQL query that checks for uniqueness case insensitive using the `LOWER()` function:
```sql
SELECT  1 AS one FROM "posts" WHERE LOWER("posts"."some_attribute") = LOWER('A VALUE') LIMIT 1
```

### Actual behavior
AR does not use the `LOWER()` function:
```sql
SELECT  1 AS one FROM "posts" WHERE "posts"."some_attribute" = $1 LIMIT $2  [["some_attribute", "A VALUE"], ["LIMIT", 1]]
```

### System configuration
**Rails version**: 5.0.0

**Ruby version**: 2.3.0

**Comment by maclover7** [comment]
Hi @nickrivadeneira! It looks like this issue has been fixed on Rails master and `5-0-stable` -- your test case passes for me on both branches. Feel free to pull directly from either branch, or wait for the next Rails version to be released (v5.0.1) to grab the fix.

**Comment by nickrivadeneira** [comment]
Thanks! I should've searched a bit better - here's the fixing PR: #26208
